File: backend/pokerogue/src/main/java/com/pokerogue/helper/ability2/dto/AbilityResponse2.java
package com.pokerogue.helper.ability2.dto;

import com.pokerogue.helper.ability2.data.Ability;

public record AbilityResponse2(String id, String koName, String description) {

    public static AbilityResponse2 from(Ability ability) {
        return new AbilityResponse2(ability.getId(), ability.getName(), ability.getDescription());
    }
}

File: android/remote/src/main/java/poke/rogue/helper/remote/dto/response/ability/AbilityResponse2.kt
package poke.rogue.helper.remote.dto.response.ability

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AbilityResponse2(
    val id: String,
    @SerialName("koName")
    val name: String,
    val description: String,
)
